### PR TITLE
[Chore] Add interoperability e2e tests for Tempo and OpenShift Service Mesh

### PR DESCRIPTION
**Pull request by IshwarKanse**
* The PR adds interoperability e2e test cases for Tempo and OpenShift Service Mesh. 
Refer [documentation link](https://docs.openshift.com/container-platform/4.15/service_mesh/v2x/ossm-observability.html#ossm-configuring-distr-tracing-tempo_observability) for configuring Distributed Tracing platform Tempo with OSSM.
```
--- PASS: chainsaw (617.47s)
    --- PASS: chainsaw/ossm-monolithic-otel (290.62s)
    --- PASS: chainsaw/ossm-tempostack (326.85s)
PASS
Tests Summary...
- Passed  tests 2
- Failed  tests 0
- Skipped tests 0
Done.
```

Add zipkin receiver support in Tempo Monolithic to work with OpenShift Service Mesh. 
https://issues.redhat.com/browse/TRACING-4129 

* Updated the monolithic-multitenancy-openshift test to print debug logs and events if it fails. The test is working locally and when we execute it from the debug CI env. However it fails during the automated CI run. Need to investigate why.

### CODE REVIEW

File: tests/e2e-openshift-ossm/ossm-tempostack/chainsaw-test.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  creationTimestamp: null
+  name: ossm-tempostack
+spec:
+  # Avoid running this test case in parellel as it uses high resources and uses static namespaces.
```
Comment (iblancasa) [inline tests/e2e-openshift-ossm/ossm-tempostack/chainsaw-test.yaml:7]:
```suggestion
  # Avoid running this test case in parallel as it uses high resources and uses static namespaces. 
```

File: tests/e2e-openshift-ossm/ossm-monolithic-otel/install-otel-collector.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+apiVersion: opentelemetry.io/v1alpha1
```
Comment (iblancasa) [inline tests/e2e-openshift-ossm/ossm-monolithic-otel/install-otel-collector.yaml:1]: What is the purpose of this OTEL Collector? I don't see where the test sends traces to this collector

Comment (IshwarKanse) [reply tests/e2e-openshift-ossm/ossm-monolithic-otel/install-otel-collector.yaml:1]: Currently Tempo Monolithic doesn't support Zipkin receiver. Thus we have to use a OTEL collector to receive traces using a Zipkin receiver and forward it to the Monolithic instance. The OTEL collector is specified in the OSSM CR. https://github.com/grafana/tempo-operator/pull/866/files#diff-f7c021630e6a1eb8ae1c18159b0000e6d5e4de555630e800ab0e6f0799a982d7R33

**Comment by IshwarKanse** [comment]
```
--- PASS: chainsaw (1010.16s)
    --- PASS: chainsaw/ossm-monolithic-otel (312.87s)
    --- PASS: chainsaw/ossm-tempostack (347.27s)
    --- PASS: chainsaw/ossm-tempostack-otel (350.02s)
PASS
Tests Summary...
- Passed  tests 3
- Failed  tests 0
- Skipped tests 0
Done.

```

File: tests/e2e-openshift-ossm/ossm-tempostack-otel/install-otel-collector.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: simplest
+  namespace: tracing-system
+spec:
+  config: |
+    receivers:
+      zipkin:
```
Comment (iblancasa) [inline tests/e2e-openshift-ossm/ossm-tempostack-otel/install-otel-collector.yaml:9]: Could you add some verification that we can receive GRPC and HTTP `otlp` traces from the workload?

Comment (IshwarKanse) [reply tests/e2e-openshift-ossm/ossm-tempostack-otel/install-otel-collector.yaml:9]: Done, the test generates traces from a project call otlp-app using telemetrygen and sends it to the OTEL collector in tracing-system to both OTLP HTTP and GRPC. Then the traces for both protocols with the servicename telemetrygen-http and telemetrygen-grpc is verified in Tempo Monolithic instance.

Comment (IshwarKanse) [reply tests/e2e-openshift-ossm/ossm-tempostack-otel/install-otel-collector.yaml:9]: https://github.com/grafana/tempo-operator/pull/866/files#diff-a84cef7557afc667fd2929ec6aae5722987b299a6a983534bcdd1f08e02c7ca1R64

Comment (IshwarKanse) [reply tests/e2e-openshift-ossm/ossm-tempostack-otel/install-otel-collector.yaml:9]: Done, added for verification of traces sent via OTLP GRPC an HTTP in TempoStack with OTEL test case as well.

**Review by iblancasa — APPROVED** [review]

**Review by andreasgerstmayr — APPROVED** [review]

**Pull request merged**